### PR TITLE
SCC-1148 | Contact preferences

### PR DESCRIPTION
**Pull request by airudah2000**

### CODE REVIEW

File: app/controllers/IndexController.scala [hunk 1]
```diff
@@ -147,24 +148,62 @@ class IndexController @Inject() (
         ).flatten
     }
 
-  private def getSettingsSummary(agentClientStatus: Boolean)
-    (implicit request: DataRequest[_]): SummaryList =
+  private def getContactPreferencesRow(implicit request: DataRequest[_], messages: Messages): Future[Option[SummaryListRow]] = {
```
Comment (owen-opencast) [inline app/controllers/IndexController.scala:151]: Could this be moved to a helper class, similar to ManageTaxAgentSummary.row or TrustedHelpersSummary.row?

File: app/config/FrontendAppConfig.scala [hunk 2]
```diff
@@ -42,6 +42,8 @@ class FrontendAppConfig @Inject() (configuration: Configuration) {
 
   val technicalSupportUrl: String = configuration.get[String]("urls.technicalSupport")
 
+  val profileAndSettingsUrl                     = s"$host/personal-account/profile-and-settings"
```
Comment (owen-opencast) [inline app/config/FrontendAppConfig.scala]: I think this is getting used to redirect back to our page? In which case the URL would be $host/hmrc-account/update-your-details

**Review by owen-opencast — CHANGES_REQUESTED** [review]

**Review by owen-opencast — COMMENTED** [review]
The unit tests need updating to pass - they shouldn't have any reliance on the stub, the stub should only be getting used in integration tests, ui tests and environments.

When unit testing we tend to mock components outside of the specific component we're testing using Mockito. So when testing the IndexController I'd mock the PreferencesConnector, and test that the correct output is rendered when certain responses are returned from preferences.

CitizenDetails is mocked in a similar manner, eg:

`val mockCitizenDetailsConnector: CitizenDetailsConnector = mock[CitizenDetailsConnector]`
`val application = applicationBuilder(userData = Some(populatedUserData), userAnswers = Some(emptyUserAnswers))
        .configure("features.sca-wrapper-enabled" -> false)
        .overrides(
         ....
          bind[CitizenDetailsConnector].toInstance(mockCitizenDetailsConnector)
        )
        .build()`

and 

`when(mockCitizenDetailsConnector.getPersonalDetails(any())(any()))
          .thenReturn(Future.successful(citizenDetailsResponseWithNoRlsFlags))`

If you need help with the tests, Ned or Christo should be able to help

File: app/config/FrontendAppConfig.scala (hunk 2 above)
Comment (airudah2000) [reply app/config/FrontendAppConfig.scala]: Fixed

File: app/connectors/HttpClientResponse.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import cats.data.EitherT
+import com.google.inject.Inject
+import play.api.Logging
+import play.api.http.Status.{BAD_GATEWAY, LOCKED, NOT_FOUND, TOO_MANY_REQUESTS}
+import uk.gov.hmrc.http.{HttpException, HttpResponse, UpstreamErrorResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class HttpClientResponse @Inject()(implicit ec: ExecutionContext) extends Logging {
```
[the remark this answers is not in the document]
Comment (airudah2000) [reply app/connectors/HttpClientResponse.scala:27]: Deleted

File: conf/application.conf [hunk 4]
```diff
@@ -51,6 +51,7 @@ play {
 play.filters.enabled += "play.filters.csp.CSPFilter"
 
 play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientModule"
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientV2Module"
```
[the remark this answers is not in the document]
Comment (airudah2000) [reply conf/application.conf:54]: Yes. Will remove

File: app/controllers/IndexController.scala (hunk 1 above)
Comment (airudah2000) [reply app/controllers/IndexController.scala:151]: Will do

File: test/controllers/IndexControllerWithoutScaWrapperLibrarySpec.scala [hunk 5]
```diff
@@ -935,7 +935,7 @@ class IndexControllerWithoutScaWrapperLibrarySpec
           "must only display certain details (name, nino, main address, postal address)" in {
```
[the remark this answers is not in the document]
Comment (airudah2000) [reply test/controllers/IndexControllerWithoutScaWrapperLibrarySpec.scala:963]: Done.

File: app/config/FrontendAppConfig.scala [hunk 6]
```diff
@@ -73,6 +79,7 @@ class FrontendAppConfig @Inject() (configuration: Configuration) {
   lazy val pertaxFrontendUrl                            = s"$pertaxFrontend/personal-account"
   lazy val pertaxFrontendMessagesUrl                    = s"$pertaxFrontend/personal-account/messages"
   lazy val pertaxFrontendProfileUrl                     = s"$pertaxFrontend/personal-account/profile-and-settings"
+  lazy val pertaxFrontendPaperlessUrl                   = s"$pertaxFrontend/paperless/check-settings?"
```
Comment (owen-opencast) [inline app/config/FrontendAppConfig.scala:82]: Is the '?' a the end of here correct?

**Review by owen-opencast — COMMENTED** [review]
Looks good, I spotted what looks like typo, but ready otherwise!

File: app/config/FrontendAppConfig.scala (hunk 6 above)
Comment (airudah2000) [reply app/config/FrontendAppConfig.scala]: No. I've removed it

**Review by owen-opencast — APPROVED** [review]

**Pull request merged**